RealmSwift/Geospatial.swift

```diff
@@ -0,0 +1,201 @@
+    ///
+    /// - Parameter latitude: Latitude in degrees. Ranges between -90 and 90 degrees, inclusive.
+    /// - Parameter longitude: Longitude in degrees. Ranges between -180 and 180 degrees, inclusive.
+    convenience init?(_ latitude: Double, _ longitude: Double) {
+        self.init(latitude: latitude, longitude: longitude)
+    }
+}
+
+/**
+ A class that represents a rectangle, that can be used in a geospatial `geoWithin`query.
+
+ - warning: This class cannot be persisted and can only be use within a geospatial `geoWithin` query.
+ */
+public typealias GeoBox = RLMGeospatialBox
+
+public extension GeoBox {
+    /// Initialize a `GeoBox`, with values for bottom left corner and top right corner.
+    ///
+    /// - Parameter bottomLeft: The bottom left corner of the rectangle.
+    /// - Parameter topRight: The top right corner of the rectangle.
+    convenience init?(bottomLeft: (Double, Double), topRight: (Double, Double)) {
+        guard let bottomLeftPoint = GeoPoint(bottomLeft.0, bottomLeft.1),
+              let topRightPoint = GeoPoint(topRight.0, topRight.1) else {
+            return nil
+        }
+        self.init(bottomLeft: bottomLeftPoint, topRight: topRightPoint)
+    }
+}
+
+/**
+ A class that represents a polygon, that can be used in a geospatial `geoWithin`query.
+
+ A `GeoPolygon` describes a shape conformed of and outer `Polygon`, called `outerRing`,
+ and 0 or more inner `Polygon`s, called `holes`, which represents an unlimited number of internal holes
+ inside the outer `Polygon`.
+ A `Polygon` describes a shape conformed by at least three segments, where the last and the first `GeoPoint`
+ must be the same to indicate a closed polygon (meaning you need at least 4 points to define a polygon).
+ Inner holes in a `GeoPolygon` must  be entirely inside the outer ring
+ 
+ A `hole` has the following restrictions:
+ - Holes may not cross, i.e. the boundary of a hole may not intersect both the interior and the exterior of any other
+   hole.
+ - Holes may not share edges, i.e. if a hole contains and edge AB, the no other hole may contain it.
+ - Holes may share vertices, however no vertex may appear twice in a single hole.
+ - No hole may be empty.
+ - Only one nesting is allowed.
+
+ - warning: This class cannot be persisted and can only be use within a geospatial `geoWithin` query.
+ */
+public typealias GeoPolygon = RLMGeospatialPolygon
+
+public extension GeoPolygon {
+    /// Initialize a `GeoPolygon`, with values for bottom left corner and top right corner.
+    ///
+    /// Returns `nil` if the `GeoPoints` representing a polygon (outer ring or holes), don't have at least 4 points.
+    /// Returns `nil` if the first and the last `GeoPoint` in a polygon are not the same.
+    ///
+    /// - Parameter outerRing: The polygon's external (outer) ring.
+    /// - Parameter holes: The holes (if any) in the polygon.
+    convenience init?(outerRing: [(Double, Double)], holes: [[(Double, Double)]]? = nil) {
```

Comment:
`holes` could be non-optional and default to [] as there isn't a semantic difference between empty and nil.